frontend/src/hooks.server.ts

```diff
@@ -57,9 +43,13 @@ export const handle: Handle = async ({ event, resolve }) => {
 	});
 
 	if (!res.ok) {
-		console.log('bad response:', await res.text());
-		// The session is invalid. Load page as normal.
-		return await resolve(event);
+		// The session is invalid. Delete the session cookie and redirect to the login page.
+		event.cookies.delete('sessionid', {
+			path: '/',
+			secure: true,
+			sameSite: 'lax'
+		});
+		redirect(302, `/login?next=${event.url.pathname}`);
```

Comment:
This is the important part, the rest of the diff is removing dead code and getting stuff out into functions